app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/dialog/DialogFactory.java

```diff
@@ -79,33 +78,57 @@ Dialog createDialog() {
     }
 
     @Nullable
-    Dialog createRangedDialog(Context context, Tag tag, EditViewModel viewModel) {
-        AlertDialog.Builder alertDialog = new AlertDialog.Builder(context);
-        alertDialog.setTitle(tag.getKey());
+    Dialog createRangedDialog(Context context, Tag baseTag, EditViewModel viewModel) {
```

Comment:
I think there is too much going on in this method. I would like you to split this method in multiple methods, or extract some code to separate classes.